head.S

```diff
@@ -164,9 +171,18 @@ GLOBAL(_entry)
 	push	$0
 	popf
 
+	cmpl	$0, boot_protocol(%ebp)
```

Comment:
`$LINUX_BOOT` according to your header (I think?)

Reply:
Done